### ISSUE TITLE
v0.0.5 Review

### ISSUE THREAD

**Issue by rickmanelius** [drud/quicksprint#21]
Might as well do a deep dive review akin to how we review/release ddev itself. See https://github.com/drud/ddev/issues/746.

Initial AC might be:

* Completely working D8 site spin up from nothing (no docker, no ddev) on the following:
  * [ ] macOS High Sierra
  * [ ] Windows 10 Pro
  * [ ] Ubuntu 16.04

**Comment by rickmanelius** [comment]
# Testing

Performed by Rick Manelius as of 2018-04-04

## macOS

### Hardware Setup

I’m using a laptop with the following high-level specifications:

* MacBook Pro: Retina 15-inch Mid 2015
* 2.2 Ghz Intel Core i7
* 16 GB 1600 MHz DDR3 RAM
* 250 GB Hard Drive

### Software Setup

* macOS Sierra 10.12.4
* Xcode
* Git 2.14.3
* Note that I specifically removed all traces of `go`, `ddev`

### Preparation


#### Cleanup

* `brew uninstall go` && `brew uninstall ddev`
* Remove "Docker for Mac" from menu and remove application from /Applications

#### Assets

* Download drupal_sprint_package.tar.gz from https://github.com/drud/quicksprint/releases/tag/v0.0.5
* Unzip from macOS finder into ~/tmp
  * Observed ~/tmp/drupal_sprint_package
  * With Term (not iTerm), cd into drupal_sprint_package

#### Install Scripts

* Run `install_ddev.sh` as per the readme.
  * Suggestion, let's specifically call out `./install_ddev.sh` for the avoidance of doubt and for copy/pasters.
  * I like "# !!You don't need to hit enter!!.", but being a rebel, I hit enter and exited. I probably should have been warned that I would need to start over because nothing happened and I could have naively assumed I was successful.
  * I hit the "y" key instead like I'm supposed to.
  * I error out with the following:
```
Installing docker images for ddev to use...
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
tar: /tmp: Not found in archive
tar: -C: Not found in archive
tar: ddev_tarballs/ddev_macos.v0.16.0.tar.gz: Not found in archive
tar: Error exit delayed from previous errors.
chmod: /tmp/ddev: No such file or directory
Ready to place ddev in your /usr/local/bin.
mv: rename /tmp/ddev to /usr/local/bin/ddev: No such file or directory
/usr/local/bin/brew
cp: /tmp/ddev_bash_completion.sh: No such file or directory
Installed ddev bash completions in /usr/local/etc/bash_completion.d
rm: /tmp/ddev_bash_completion.sh: No such file or directory
```
* It appears that the Docker for Mac uninstaller didn't properly remove docker/docker-compose from /usr/local/bin/docker. I remove these two and then re-run ./install_ddev.sh. I press `y` again and now I make it to the prompt of installing docker. I'm given the option of not installing by hitting enter instead of 'y'. I do like this delineation because now I know which way to go. It would probably be more accurate to say "Installing Docker for Mac" because it is possible to install Docker Toolbox. Alas...
* I see "Please open Docker preferences". This is probably obvious, but we may want to specify clicking on the menu icon to get to the preferences. I'm not sure if these instructions are OS specific, so I'm simply going to proceed with most users will understand this.
* I note now that we go back to the verbiage "!!You don't need to hit enter!!." It appears that in some context that hitting enter is the valid approach to exit and y is to proceed. I think most people are going to want to proceed 
* I see the Docker images are done and now I'm promped to run `cd ~/Sites/sprint` and `./start_clean.sh`, which I do and I see a lot of Drupal 8 activity. I now see `cd sprint-20180404-2359` and `./start_clean.sh`
* I perform and I'm now seeing "# This simple script starts a clean instance of drupal # running in ddev and imports a starter database."
* Note that I see ./start_clean.sh: line 33: ddev: command not found
  * The ddev binary is not installed. I'm using zsh and it's not seen in my path.
  * Attemting to re-run ./install_ddev.sh results in an issue
```
Installing docker images for ddev to use...
ls: ddev_tarballs/ddev_docker_images*.tar.xz: No such file or directory
gzip: standard input is a terminal -- ignoring
open /var/lib/docker/tmp/docker-import-010004105/repositories: no such file or directory
ls: ddev_tarballs/ddev_macos*.tar.gz: No such file or directory
tar: Error opening archive: Failed to open '-C'
chmod: /tmp/ddev: No such file or directory
Ready to place ddev in your /usr/local/bin.
mv: rename /tmp/ddev to /usr/local/bin/ddev: No such file or directory
/usr/local/bin/brew
cp: /tmp/ddev_bash_completion.sh: No such file or directory
Installed ddev bash completions in /usr/local/etc/bash_completion.d
rm: /tmp/ddev_bash_completion.sh: No such file or directory
cp: sprint.tar.xz: No such file or directory
```
  * I am able to see ddev_tarballs/ddev_macos.v0.16.0.tar.gz

I'll pick back up in the morning with another end to end after I submit a PR for some of the small docs changes and leave the "y" versus "enter" as a default recommendation up for discussion. One of the


### Recommendations

* Post Youtube video to README.md
* Link to Nashville Core Sprints page (https://events.drupal.org/nashville2018/sprints) from README.md
* s/ddev by Drud Technology/DDEV-Local by DRUD Tech (recent branding touch ups)
* s/phpmyadmin/phpMyAdmin/ to match their branding.
* s/'# !!You don't need to hit enter!!.'/'# !!If you just hit enter, you'll exit the install process!!.'/
* Perhaps instead of asking for users to hit "y" instead of enter, maybe we ask them to hit enter and (if they prefer exiting) to hit "n"? That way the simplest path forward it just keep hitting enter :) I guess technically keep hitting y also works... but a user would have to consciously exit whereas now they can accidentily exit the installer.
* It seems the macOS install assumes homebrew, but it wasn't caught in the installer.

**Comment by rickmanelius** [comment]
Also noting that 17.09.1-ce-mac42 is installed whereas 18.03 has felt more stable from my testing.

**Comment by rfay** [comment]
Yeah, the install should be 18.03. If updating the docker installs, it would be great to add the version of the install into the filename, so one doesn't have to install it to find out what version it is.

**Comment by dsquaredb** [comment]
# Testing 

Performed by Danita Bowman as of 2018-04-05

## Hardware Setup

I’m using a laptop with the following specifications:

MacBook Pro: Retina 15-inch Mid 2015

- 2.8 Ghz Intel Core i7
- 16 GB 1600 MHz DDR3 RAM
-  macOS Sierra 10.12.6
- Docker v18.03.0-ce-mac58

## Preparation

Cleanup

Ran the following commands:
-     brew uninstall ddev
-     brew uninstall -force ddev
( I was prompted to run this second command after the first uninstall)

- Deleted my old sprint site directory and deleted .ddev/sprint-20180404-1057 directory from a previous install

## Install Sprint Package

- Downloaded drupal_sprint_package.tar.gz from https://github.com/drud/quicksprint/releases/tag/v0.0.5
- Unzipped from macOS finder into ~/Downloads
- Observed ~/Downloads/drupal_sprint_package

- Moved drupal_sprint_package to a ddev directory I already have for ddev instances

- cd into ddev/drupal_sprint_package

- Looked at README.md for quicksprint on github for info on how to install. Don't run scripts, so was not sure how to follow those instructions

- Looked at SPRINTUSER_README.md in the drupal_sprint_package. Since I wasn't familiar with running script packages, I thought I was supposed to change to a './install_ddev.sh' directory so tried that and, of course, got a message that that was not a directory.

- Not to be defeated, I Googled how to run script packages in macOS and followed this: https://apple.stackexchange.com/questions/235128/how-do-i-run-a-sh-or-command-file-in-terminal and ran this: ````sh install_ddev.sh````


## Install Scripts

- Script ran and when it stopped, I hit 'Enter' by mistake and had to run the install command again.
- Second time, I hit the "y" key instead.
- Each time the script finished a section, I followed the instructions being careful not to hit 'Enter' again.

When the scripts finished, I used the access information provided to navigate to the website successfully and was able to log in (using the provided credentials) and navigate around the "Site-Install" website.

## Recommendations

Having participated in sprints before, including first-time sprinters orientations, I have seen the various experience levels of participants. Some are not familiar with terminals/command line. Others, like me, are comfortable with command line, but not as familiar with running scripts. I would recommend adding just a bit more instruction about running the script in the readme.

## Note

In Slack last night, I was asked if I had increased memory as a possible reason for my first install attempt not being totally successful. I did not make this change before attempting this install.

**Comment by rickmanelius** [comment]
I just talked with @rfay. It's likely that I broke something when removing Docker for Mac and it not happening cleanly, which may be why the ddev binary wasn't install.

**Comment by BrianGilbert** [comment]
@rfay I can't see any way top programatically set the docker install file name with the version.. it's not part of the URL, will have to manually change it every time it's updated..?

**Comment by rfay** [comment]
@BrianGilbert Please use a $DOCKER_VERSION environment variable and then just inject that into the destination filename. WIth curl I think you would use -O

**Comment by lglynn2299** [comment]
Performed by Leslie Glynn on 2018-04-05

**Hardware Setup**
I’m using a laptop with the following specifications:
MacBook Pro (15-inch, 2017)
2.8 GHz Intel Core i7
16 GB 2133 MHz LPDDR3
macOS Sierra 10.13.3

**Environment**
As a freelancer I have lots of development environments installed for various clients:  vagrant machine running on virtualbox, docksal with docker, mamp, etc.
Docker was installed, but was not running.  vm machine using vagrant was running
Homebrew was installed

**Install Sprint Package**
Instruction #1 - Extract the tar.gz file
Downloaded drupal_sprint_package.tar.gz from https://github.com/drud/quicksprint/releases/tag/v0.0.5

Unzipped from macOS finder into ~/Downloads

cd into drupal_sprint_package

Instruction #2 - Run install_ddev.sh and follow prompts
Since I wasn't familiar with running script packages, I tried "run install_ddev.sh" per the instructions and then "install_ddev.sh", both without success

I asked a devops friend who told me to add the ./ and then tried  ./install_ddev.sh

**Running Install Scripts**

When the script got to the install docker step I got this error and the script stopped:

        yChecksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified   CRC32 $09DFE74F
Checksumming GPT Header (Primary GPT Header : 1)…
 GPT Header (Primary GPT Header : 1): verified   CRC32 $6FED0F84
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified   CRC32 $DFA9650D
Checksumming  (Apple_Free : 3)…
                    (Apple_Free : 3): verified   CRC32 $00000000
Checksumming disk image (Apple_HFS : 4)…
.........................................................................
          disk image (Apple_HFS : 4): verified   CRC32 $32D498D6
Checksumming  (Apple_Free : 5)…
..............................................................................
                    (Apple_Free : 5): verified   CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified   CRC32 $DFA9650D
Checksumming GPT Header (Backup GPT Header : 7)…
  GPT Header (Backup GPT Header : 7): verified   CRC32 $458D0FC6
verified   CRC32 $02F05A5A
/dev/disk2          	GUID_partition_scheme          	
/dev/disk2s1        	Apple_HFS                      	/Volumes/Docker
No application knows how to open .
"disk2" unmounted.
"disk2" ejected.

I ran the script a 2nd time and this time I got:
        yexpected   CRC32 $02F05A5A
/dev/disk2          	GUID_partition_scheme          	
/dev/disk2s1        	Apple_HFS                      	/Volumes/Docker
"disk2" unmounted.
"disk2" ejected.

I also got a popup telling me docker was corrupt and asking to delete it.
Since I didn't know what impact removing docker would have on my other work environments I had to wait for a dev ops expert to check things out for me and tell me it was safe to delete

I ran the script a 3rd time, still getting these messages after the install docker step
        yexpected   CRC32 $02F05A5A
/dev/disk2          	GUID_partition_scheme          	
/dev/disk2s1        	Apple_HFS                      	/Volumes/Docker
"disk2" unmounted.
"disk2" ejected.

This time Docker opened though and I was able to update the memory limit per the instructions

Somewhere (sorry, don't remember if it was before or after docker started)
I got the permissions error saying I was unable to run an app based on my permissions
I had to go into settings - security and privacy and allow the app to run

I finally made it to the point where it asked me to run ./start_sprint.sh
I did that successfully and then ran ./start_clean.sh

I got these messages
A new update is available! please visit https://github.com/drud/ddev/releases/tag/v0.17.0 to download the update.
For upgrade help see https://ddev.readthedocs.io/en/latest/#installation 
Project is not currently running. Try 'ddev start'. 

since the last line above was in red and no other messages followed, I figured the script failed again

After several minutes the script continued and went down to the command Creating ddev-router

I then got these messages in red
Failed to start sprint-20180405-1700: web service health check timed out 
Project is stopped. Run 'ddev start' to start the environment. 
Project is stopped. Run 'ddev start' to start the environment. 
Project is stopped. Run 'ddev start' to start the environment. 
Project is stopped. Run 'ddev start' to start the environment. 
Project is stopped. Run 'ddev start' to start the environment. 

The link to the website was displayed as well, but going there (http://sprint-20180405-1700.ddev.local/) gave me a 404

**Thoughts**
Thanks for all the hard work that has gone into and will continue to go into making this a great tool. Having mentored at many sprints before and having run several "Get involved with core contributing for first-time sprinters" I have seen the various experience levels of participants as well as the unique configurations and permission lockdowns that they bring with them. Having all kinds of combinations of already installed Drupal dev environments (Docksal, Lando, Mamp, etc) in todays Drupal environment make things even more challenging.

I have no specific recommendations that others haven't already added to this thread, but am concerned with the issues we will face using this tool at the first-time sprinters workshop in Nashville.  Let me know how I can help your team before the sprints.

**Comment by rfay** [comment]
I think there was something wrong with the packaged docker install for macos. I couldn't get it to work at all either.

**Comment by BrianGilbert** [comment]
I think I will remove the docker installers from quicksprint, and include them in a separate folder of what is supplied for a drupalcon sprint.

**Comment by BrianGilbert** [comment]
Reports addressed with PR for 0.0.6
